language-support/scala/codegen/src/main/scala/com/digitalasset/codegen/CodeGen.scala

```diff
@@ -75,18 +80,20 @@ object CodeGen {
       files: NonEmptyList[File],
       packageName: String,
       outputDir: File,
-      mode: Mode): ValidationNel[String, Unit] =
+      mode: Mode,
+      roots: Seq[String]): ValidationNel[String, Unit] =
     decodeInterfaces(files).map { ifaces: NonEmptyList[EnvironmentInterface] =>
       val combinedIface: EnvironmentInterface = combineEnvInterfaces(ifaces)
```

Comment:
To explain further about doing the filtering outside of the dependency graph code, while dealing with the case @jberthold-da mentioned about a template referring to another template's associated record type as a normal data structure:

```suggestion
       val combinedIface: EnvironmentInterface = combineEnvInterfaces(ifaces map (filterTemplatesByRegexes(roots, _)))
```
And elsewhere, something like

```scala
private[this] def filterTemplatesByRegexes(regexes: Seq[String], ei: EnvironmentInterface): EnvironmentInterface =
  if (regexes.isEmpty) ei else {
    val EnvironmentInterface(tds) = ei
    EnvironmentInterface(tds transform {
      case (id, tpl@InterfaceType.Template(_, _)) if matchesRoots(id) => // matchesRoots from your existing code
        InterfaceType.Normal(tpl.`type`)
      case (_, other) => other
    })
  }
```

This has the nice effect of generating those indirectly included templates using only their "normal datatype" form.